src/fileio.cpp

```diff
@@ -279,7 +279,7 @@ bool FioCheckFileExists(const std::string &filename, Subdirectory subdir)
  */
 bool FileExists(const std::string &filename)
 {
-	return access(OTTD2FS(filename.c_str()), 0) == 0;
+	return access(OTTD2FS(filename).c_str(), 0) == 0;
```

Comment:
I'm concerned about the correctness of `OTTD2FS(..).c_str()` and it's use of the temporary std::string object. Is that definitely all correct?

Reply:
In this case the temporary will not be destructed until the end of the statement (after access has returned), so it is OK.

Reply:
How about all the others? :)

Reply:
I went over the changes and checked, yes all the cases of `OTTD2FS(x).c_str()` are safe: The C string returned is only expected to live until the called function has returned. All the cases where it caused problems the temporary string returned from the conversion function is stored into a proper local variable to manage its lifetime.